src/EditorFeatures/Core/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -44,7 +46,21 @@ internal ItemManager(RecentItemsManager recentItemsManager, EditorOptionsService
             CancellationToken cancellationToken)
         {
             var stopwatch = SharedStopwatch.StartNew();
-            var items = SortCompletionItems(data, cancellationToken).ToImmutableArray();
+
+            var list = s_sortListPool.Allocate();
+            ImmutableArray<VSCompletionItem> items;
+
+            try
+            {
+                SortCompletionItems(list, data, cancellationToken);
+
+                items = list.ToImmutableArray();
```

Comment:
This path isn't executed in VS, it only exists for back-compat. But if this to be executed, given that ArrayBuilder and List has similar sorting performance, it might be better to stick to ArrayBuilder here to avoid since we are calling `ToImmutableArray` (and allocate) anyway.

Again, probably doesn't matter :)

Reply:
Yeah, I definitely didn't see this codepath exercised when debugging VS. I'd prefer to keep this method similar to SortCompletionItemListAsync, unless there is a good reason not to. 

Either way, I'd like to better understand the perf concern you have here. Ignoring the pooled array allocations, from what I see, this method would only allocate once (creating the immutable array from the list), no? SortCompletionItemListAsync has the advantage of no allocations (in our code, platform would still need to allocate a list from the IEnumerable we send them).

Reply:
Well, I feel like I'm wasting your time commenting on code not going to be executed. But my point is, in `SortCompletionListAsync`, using an ArrayBuilder directly would have the same allocation as using a pooled List then calling `ToImmutableArray`, but avoid the copy and clear steps.

> platform would still need to allocate a list from the IEnumerable we send them

Yea, this is done this way, i.e. `session.CreateCompletionList(list)`, just so we can use pooled list owned by editor to avoid LOH allocation.